src/actions/exchangeActions.js

```diff
@@ -118,81 +117,52 @@ export const takeOfferAction = (
       return;
     }
 
-    let { address: fromAssetAddress } = fromAsset;
+    const { address: fromAssetAddress } = fromAsset;
     const { decimals: fromAssetDecimals } = fromAsset;
-    let { address: toAssetAddress } = toAsset;
-
-    // we need PROD assets' addresses in order to get offers when on ropsten network
-    // as v2 requests require to provide addresses not tickers
-    if (__DEV__) {
-      const prodAssetsAddress = await exchangeService.getProdAssetsAddress();
-      fromAssetAddress = prodAssetsAddress[fromAssetCode];
-      toAssetAddress = prodAssetsAddress[toAssetCode];
-    }
 
-    const offerRequest = {
-      quantity: parseFloat(fromAmount),
-      provider,
-      fromAssetAddress,
-      toAssetAddress,
-      walletId: activeWalletId,
-    };
-    const order = await exchangeService.takeOffer(offerRequest, trackId);
-    const offerOrderData = get(order, 'data');
-    if (isEmpty(offerOrderData) || order.error) {
-      let { message = 'Unable to request offer' } = order.error || {};
-      if (message.toString().toLowerCase().includes('kyc')) {
-        message = 'Shapeshift KYC must be complete in order to proceed';
-      }
-      Toast.show({
-        title: 'Exchange service failed',
-        type: 'warning',
-        message,
-      });
-      callback({}); // let's return callback to dismiss loading spinner on offer card button
-      return;
-    }
-    const {
-      payToAddress,
-      payQuantity,
-    }: OfferOrder = offerOrderData;
-    const transactionDataString = get(offerOrderData, 'transactionObj.data');
-
-    const from = getActiveAccountAddress(accounts);
-    const transactionPayload = {
-      from,
-      to: payToAddress,
-      data: transactionDataString,
-      amount: payQuantity,
-      symbol: fromAssetCode,
+    const transactionData = {
+      fromAsset,
+      toAsset,
+      from: getSmartWalletAddress(accounts),
+      payQuantity: fromAmount,
+      amount: fromAmount,
+      symbol: fromAsset.symbol,
       contractAddress: fromAssetAddress || '',
       decimals: parseInt(fromAssetDecimals, 10) || 18,
+      ...order.transactionObj,
+
     };
-    callback({
-      ...offerOrderData,
-      transactionPayload,
-    });
+    callback(transactionData);
   };
 };
 
 export const resetOffersAction = () => {
-  return async (dispatch: Dispatch) => {
-    // reset websocket listener
-    exchangeService.resetOnOffers();
+  return (dispatch: Dispatch) => {
     dispatch({ type: RESET_OFFERS });
   };
 };
```

Comment:
Can be done as:

`export const resetOffersAction = () = ({ type: RESET_OFFERS });`